app/controllers/api/v1/repos_controller.rb

```diff
@@ -32,20 +32,30 @@ def show
   # POST /api/v1/users/:user_id/repos
   def create
     user = User.find(params[:user_id])
-    repo = user.repos.new(repo_params)
 
-    if repo.save
-      render json: repo, status: :created
+    if user.repos.exists?(owner: repo_params[:owner], name: repo_params[:name])
+      render json: { error: "Repo already exists." }, status: :unauthorized
     else
-      render json: repo.errors, status: :unauthorized
+      repo = user.repos.new(repo_params)
+
+      if repo.save
+        render json: repo, status: :created
+      else
+        render json: repo.errors, status: :unauthorized
+      end
     end
   end
 
   # DELETE /api/v1/users/:user_id/repos/:id
   def destroy
     user = User.find(params[:user_id])
     repo = user.repos.find(params[:id])
-    repo.destroy!
+    
+    if repo.nil?
+      render json: { error: "Repo not found." }, status: :not_found
+    else
+      repo.destroy!
+    end
   end
```

Comment:
Nice and clean. I noticed that we are switching between using numbers (404) vs symbols (:not_found). If there is time, I would like to make them consistent. (I think it is mostly symbols at this point.)